app/(route)/join/page.tsx

```diff
@@ -7,140 +7,141 @@ import React from 'react';
-                    <Button
-                        size="small"
-                        onClick={checkEmail}
-                        styles={{
-                            height: '3.8rem',
-                            marginTop: '3.6rem',
-                            marginLeft: '1rem',
-                        }}
-                    >
-                        확인
-                    </Button>
-                </EmailFormWrapper>
-                <Input
-                    type="text"
-                    label="이름*"
-                    name="username"
-                    placeholder="성함을 입력해주세요"
-                    errormessage={errors.username?.message || ''}
-                    register={register}
-                    rules={{
-                        required: true,
-                        pattern: {
-                            value: /^[가-힣]+$/,
-                            message: '입력하신 성함이 올바르지 않습니다.',
-                        },
-                    }}
-                ></Input>
-                <Button type="submit" styles={{ marginTop: '2rem' }}>
-                    가입하기
-                </Button>
-                <Button buttonColor="light" onClick={() => router.push('/')}>
-                    돌아가기
-                </Button>
-            </form>
-        </JoinContainer>
-    );
+  const {
+    register,
+    handleSubmit,
+    getValues,
+    formState: { errors },
+  } = useForm<FormData>({
+    mode: 'onChange',
+  });
+  const onSubmit = (data: FormData) => {
+    //data submit
+  };
+  const checkEmail = () => {
+    const email = getValues('email');
+    //email 체크
+  };
+  const router = useRouter();
+  return (
+    <JoinContainer>
+      <Title>회원가입</Title>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <Input
+          type="text"
+          label="아이디*"
+          name="id"
+          placeholder="아이디를 입력해주세요"
+          errormessage={errors.id?.message || ''}
+          register={register}
+          rules={{
+            pattern: {
+              value: /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/,
+              message: '올바르지 않은 형식입니다.',
+            },
+          }}
+        ></Input>
+        <Input
+          type="password"
+          label="비밀번호*"
+          name="password"
+          placeholder="비밀번호를 입력해주세요"
+          errormessage={errors.password?.message || ''}
+          register={register}
+          rules={{
+            pattern: {
+              value:
+                /^(?=.*[a-zA-Z])(?=.*\d)(?=.*[!@#$%^&*()\-_=+\\\|\[\]{};:'",.<>\/?]).{8,16}$/,
+              message:
+                '영문, 숫자, 특수문자를 조합해서 입력해주세요. (8~16글자)',
+            },
+            maxLength: {
+              value: 16,
+              message:
+                '영문, 숫자, 특수문자를 조합해서 입력해주세요. (8~16글자)',
+            },
+            minLength: {
+              value: 8,
+              message:
+                '영문, 숫자, 특수문자를 조합해서 입력해주세요. (8~16글자)',
+            },
+          }}
```

Comment:
```suggestion
            required: {
              value: true,
              message: '필수 값입니다.',
            },
            pattern: {
              value:
                /^(?=.*[a-zA-Z])(?=.*\d)(?=.*[!@#$%^&*()\-_=+\\\|\[\]{};:'",.<>\/?]).{9,}$/,
              message: '영문, 숫자, 특수문자를 조합해서 9자 이상 입력해주세요. (대소문자 구분)',
            },
            minLength: {
              value: 9,
              message: '영문, 숫자, 특수문자를 조합해서 9자 이상 입력해주세요. (대소문자 구분)',
            },
          }}
```

비밀번호 rule 수정되었습니다.
더 좋은 message가 있다면 수정하셔도 됩니다.